src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -714,6 +705,33 @@ public MemoryPoolIterator CopyTo(byte[] array, int offset, int count, out int ac
             }
         }
 
+        public MemoryPoolIterator Consume(int count, out int actual)
+        {
+            var block = _block;
+            var index = _index;
+            var remaining = count;
+            while (true)
+            {
+                var following = block.End - index;
```

Comment:
While you might not need to use the `wasLastBlock` pattern since we want to skip it anyway, I don't feel to good about using a different pattern for traversing blocks in this one place.

It certainly risks returning an invalid `actual` out param. While this might never technically be used today by callers to `CopyTo` with a null array, I don't think it's OK for it to ever be wrong in case it is used.